src/core/handle.ml

```diff
@@ -25,7 +25,8 @@ let check_builtin_nat : popt -> sym StrMap.t -> string -> sym -> unit
      let symb_0 = builtin "0" in
      let typ_0 = !(symb_0.sym_type) in
      let x = Bindlib.new_var mkfree "_" in
-     let typ_s = Ctxt.to_prod [(x, typ_0)] typ_0 in
+     let ctx = [(x, typ_0, None)] in
+     let typ_s = Ctxt.to_prod ctx typ_0 in
```

Comment:
ctx can be inlined

Reply:
In fact, it would be better to introduce in terms.ml a function
```
let imply t u =
  let dummy = Bindlib.new_var mkfree "_" in
  Prod (t, Bindlib.bind_var dummy (Bindlib.box u))
```

Reply:
not quite

Reply:
I wrote `imply` into `basics`